### PR TITLE
Enable cheats over multiplayer and fix some desync causes

### PR DESCRIPTION
**Pull request by zsilencer**

### CODE REVIEW

**Comment by janisozaur** [comment]
Can you please bump `NETWORK_STREAM_VERSION` in https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/network/network.h#L71? This is precisely the kind of change it was meant for.

File: src/peep/peep.c [hunk 1]
```diff
@@ -7206,7 +7206,7 @@ static uint16 sub_69A997(sint16 x, sint16 y, uint8 z, uint8 counter, uint16 scor
 
 	do {
 		edges &= ~(1 << test_edge);
-		int saved_f1aedc = RCT2_GLOBAL(0x00F1AEDC, int);
+		sint8 saved_f1aedc = RCT2_GLOBAL(0x00F1AEDC, sint8);
```
Comment (duncanspumpkin) [inline src/peep/peep.c]: Why is this getting changed?

Comment (zsilencer) [inline src/peep/peep.c]: ok, i will change it to RCT2_ADDRESS(int) to make that more clear.

File: src/network/network.cpp [hunk 2]
```diff
@@ -1671,62 +1680,55 @@ void Network::Client_Handle_MAP(NetworkConnection& connection, NetworkPacket& pa
...
+	memcpy(&chunk_buffer[offset], (void*)packet.Read(chunksize), chunksize);
+	if (offset + chunksize == size) {
+		window_network_status_close();
+		bool has_to_free = false;
+		unsigned char *data = &chunk_buffer[0];
+		size_t data_size = size;
+		// zlib-compressed
+		if (strcmp("open2_sv6_zlib", (char *)&chunk_buffer[0]) == 0)
+		{
+			log_warning("Received zlib-compressed sv6 map");
```
Comment (Gymnasiast) [inline src/network/network.cpp]: Would a log_verbose be a better fit here (and at some other places)?

Comment (janisozaur) [inline src/network/network.cpp]: Yes, indeed. I originally left it by mistake here while doing some testing and forgot to switch it back to `log_verbose`.

**Comment by janisozaur** [comment]
Tested most of it, looks good!

**Comment by Gymnasiast** [comment]
Nice work!

**Pull request merged**